src/wallet/wallet.cpp

```diff
@@ -1846,35 +1862,37 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
         uint256 next_block_hash;
         chain().findBlock(block_hash, FoundBlock().inActiveChain(block_still_active).nextBlock(FoundBlock().inActiveChain(next_block).hash(next_block_hash)));
 
-        if (!block.IsNull()) {
-            LOCK(cs_wallet);
-            if (!block_still_active) {
-                // Abort scan if current block is no longer active, to prevent
-                // marking transactions as coming from the wrong block.
-                result.last_failed_block = block_hash;
-                result.status = ScanResult::FAILURE;
-                break;
-            }
-            for (size_t posInBlock = 0; posInBlock < block.vtx.size(); ++posInBlock) {
-                SyncTransaction(block.vtx[posInBlock], TxStateConfirmed{block_hash, block_height, static_cast<int>(posInBlock)}, fUpdate, /*rescanning_old_block=*/true);
-            }
-            // scan succeeded, record block as most recent successfully scanned
-            result.last_scanned_block = block_hash;
-            result.last_scanned_height = block_height;
+        if (!skip_block) {
```

Comment:
Doesn't ` if (!skip_block) chain().findBlock(block_hash, FoundBlock().data(block));` guarantee that the block will be null if `skip_block` is true ?

Reply:
Note that the `if (!block.IsNull()) { ... }` block has an else-condition below that we don't want to reach (it sets `ScanResult::FAILURE`) if we skipped the block due to the fast rescan filter. Reviewing with `--ignore-all-space` should show the diff a bit more clearly to see it. (Or did I interpret your comment wrongly and you meant something else?)

Reply:
Got it. Originally if the block is null, this is a reason for failure. With `skip_block`, this can happen if the filter was applied.